cudaextender/samples/sample_cudaextender.cpp

```diff
@@ -0,0 +1,182 @@
+                                                                  stream0.get(),
+                                                                  0,
+                                                                  allocator);
+    if (!device_ptr_api_mode)
+    {
+        ungapped_extender->extend_async(h_encoded_query.data(),
+                                        h_encoded_query.size(),
+                                        h_encoded_target.data(),
+                                        h_encoded_target.size(),
+                                        score_threshold,
+                                        h_seed_pairs);
+        ungapped_extender->sync();
+        h_ssp = ungapped_extender->get_scored_segment_pairs();
+    }
+    else
+    {
+        // Allocate space on device for target and query sequences, seed_pairs,
+        // scored segment pairs (ssp) and num_ssp using default allocator (caching)
+        // Allocate space for query and target sequences
+        device_buffer<char> d_query(query_sequence.length(), allocator, stream0.get());
+        device_buffer<char> d_target(target_sequence.length(), allocator, stream0.get());
+        // Allocate space for SeedPair input
+        device_buffer<SeedPair> d_seed_pairs(h_seed_pairs.size(), allocator, stream0.get());
+        // Allocate space for ScoredSegmentPair output
+        device_buffer<ScoredSegmentPair> d_ssp(h_seed_pairs.size(), allocator, stream0.get());
+        device_buffer<int32_t> d_num_ssp(1, allocator, stream0.get());
+
+        // Async Memcopy all the input values to device
+        device_copy_n(h_encoded_query.data(), query_sequence.length(), d_query.data(), stream0.get());
+        device_copy_n(h_encoded_target.data(), target_sequence.length(), d_target.data(), stream0.get());
+        device_copy_n(h_seed_pairs.data(), h_seed_pairs.size(), d_seed_pairs.data(), stream0.get());
+
+        // Launch the ungapped extender device pointer function
+        ungapped_extender->extend_async(d_query.data(),
+                                        d_query.size(),
+                                        d_target.data(),
+                                        d_target.size(),
+                                        score_threshold,
+                                        d_seed_pairs.data(),
+                                        d_seed_pairs.size(),
+                                        d_ssp.data(),
+                                        d_num_ssp.data());
+
+        // Wait for ungapped extender to finish
+        GW_CU_CHECK_ERR(cudaStreamSynchronize(stream0.get()));
+        const int32_t h_num_ssp = cudautils::get_value_from_device(d_num_ssp.data(), stream0.get());
+        h_ssp.resize(h_num_ssp);
+        // Copy data synchronously
+        device_copy_n(d_ssp.data(), h_num_ssp, h_ssp.data());
+    }
+    std::cerr << "Number of Scored Segment Pairs found: " << h_ssp.size() << std::endl;
+    if (print)
```

Comment:
minor - can you rewrite this as
```
if (print)
{
    ...
}
```

I thought our linter would catch this...

Reply:
Done.